### PR TITLE
K8s/Folders: Add basic folder integration tests (create, read)

### PR DESCRIPTION
**Pull request by suntala**
Fixes https://github.com/grafana/grafana-org/issues/202

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/tests/apis/folder/folders_test.go [hunk 1]
```diff
@@ -86,4 +102,268 @@ func TestIntegrationFoldersApp(t *testing.T) {
...
+				featuremgmt.FlagKubernetesFolders,
+			},
+		}))
+	})
+
+	t.Run("with dual write (unified storage, mode 0)", func(t *testing.T) {
+		doFolderTests(t, apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction:    false,
+			DisableAnonymous:     true,
+			APIServerStorageType: "file", // write the files to disk
```
Comment (filewalkwithme) [inline pkg/tests/apis/folder/folders_test.go:155]: This should be using unified-storage, no?

File: pkg/tests/apis/folder/folders_test.go [hunk 2]
```diff
@@ -86,4 +102,268 @@ func TestIntegrationFoldersApp(t *testing.T) {
...
+				featuremgmt.FlagKubernetesFolders,
+			},
+		}))
+	})
+
+	t.Run("with dual write (unified storage, mode 1)", func(t *testing.T) {
+		doFolderTests(t, apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction:    false,
+			DisableAnonymous:     true,
+			APIServerStorageType: "file", // write the files to disk
```
Comment (filewalkwithme) [inline pkg/tests/apis/folder/folders_test.go:172]: This should be using unified-storage, no?

File: pkg/tests/apis/folder/folders_test.go (hunk 1 above)
Comment (suntala) [reply pkg/tests/apis/folder/folders_test.go]: True! Thanks for catching that.

**Review by filewalkwithme — APPROVED** [review]
:+1:

**Pull request merged**